src/raps/actions/crosschainSwap.ts

```diff
@@ -125,10 +187,11 @@ export const crosschainSwap = async ({
 
   let gasLimit;
   try {
-    gasLimit = await estimateCrosschainSwapGasLimit({
-      chainId,
-      requiresApprove,
+    gasLimit = await estimateUnlockAndCrosschainSwap({
```

Comment:
main change is to use this function instead, which takes into account the metadata estimations